samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -206,9 +243,174 @@ def _translate_to_cfn(tf_json: dict) -> dict:
+        resource_type = (
+            sam_metadata_resource.sam_metadata_resource.get("values", {}).get("triggers", {}).get("resource_type")
+        )
+        sam_metadata_resource_address = sam_metadata_resource.sam_metadata_resource.get("address")
+        enrichment_function = resources_types_enrichment_functions.get(resource_type)
+        if not enrichment_function:
+            raise InvalidSamMetadataProperties(
+                f"The resource type {resource_type} found in the sam metadata resource {sam_metadata_resource_address} "
+                f"is not a correct resource type. The resource type should be one of these values "
+                f"{resources_types_enrichment_functions.keys()}"
+            )
+
+
+def _get_relevant_cfn_resource(
+    sam_metadata_resource: SamMetadataResource, cfn_resources: Dict[str, Dict]
+) -> Tuple[Dict, str]:
+    """
+    use the sam metadata resource name property to determine the resource address, and transform the address to logical
+    id to use it to get the cfn_resource.
+
+    Parameters
+    ----------
+    sam_metadata_resource: SamMetadataResource
+        sam metadata resource that contain extra information about some resource.
+    cfn_resources: Dict
+        CloudFormation resources
+
+    Returns
+    -------
+    tuple(Dict, str)
+        The cfn resource that mentioned in the sam metadata resource, and the resource logical id
+    """
+    sam_metadata_resource_address = sam_metadata_resource.sam_metadata_resource.get("address")
+    resource_name = (
+        sam_metadata_resource.sam_metadata_resource.get("values", {}).get("triggers", {}).get("resource_name")
+    )
+    if not resource_name:
+        raise InvalidSamMetadataProperties(
+            f"sam cli expects the sam metadata resource {sam_metadata_resource_address} to contain a resource name "
+            f"that will be enriched using this metadata resource"
+        )
+
+    # the provided resource name can be the name without the module address, or can be a complete resource address name
+    # check first if the provided name is without the module address
+    LOG.info(
+        "Check if the input source name %s is a postfix to the current module address %s",
+        resource_name,
+        sam_metadata_resource.current_module_address,
+    )
+    full_resource_address = (
+        f"{sam_metadata_resource.current_module_address}.{resource_name}"
+        if sam_metadata_resource.current_module_address
+        else resource_name
+    )
+    LOG.debug("check if the resource address %s has a relevant cfn resource or not", full_resource_address)
+    logical_id = _build_cfn_logical_id(full_resource_address)
+    cfn_resource = cfn_resources.get(logical_id)
+    if cfn_resource:
+        LOG.info("The CFN resource that match the input resource name %s is %s", resource_name, logical_id)
+        return cfn_resource, logical_id
+
+    LOG.info("Check if the input source name %s is a complete address", resource_name)
```

Comment:
I think let's change `source` to `resource`, to be consistent with the other info messages here.

Reply:
Done